### PR TITLE
B2

### PR DESCRIPTION
**Pull request by grafikrobot**
Specify library name and version:  **b2/4.x.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

This is a refactor/cleanup of the existing B2 versions to prepare for the 4.3.0 release package. Doing this as two PRs as the CCI checks don't allow otherwise.

### CODE REVIEW

File: recipes/b2/standard/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    '''
+    options = {
+        'use_cxx_env': [False, True],
+        'toolset': [
+            'auto', 'cxx', 'cross-cxx',
+            'acc', 'borland', 'clang', 'como', 'gcc-nocygwin', 'gcc',
+            'intel-darwin', 'intel-linux', 'intel-win32', 'kcc', 'kylix',
+            'mingw', 'mipspro', 'pathscale', 'pgi', 'qcc', 'sun', 'sunpro',
+            'tru64cxx', 'vacpp', 'vc12', 'vc14', 'vc141', 'vc142' ]
+    }
```
Comment (danimtb) [inline recipes/b2/standard/conanfile.py:42]: I think this options should be removed in the package_id() method as those only inject different flags to the b2 command line, but do not generate a different binary for b2. In other words, those options only affect the compilation of the consumers

**Comment by uilianries** [comment]
@grafikrobot Is there any reason for renaming `all` to `standard`? I would prefer keeping `all`, because almost all recipes are using it.

File: recipes/b2/standard/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    default_options = {
+        'use_cxx_env': False,
+        'toolset': 'auto'
+    }
+
+    def configure(self):
+        if self.options.toolset == 'cxx' or self.options.toolset == 'cross-cxx':
+            self.options.use_cxx_env = True
```
Comment (uilianries) [inline recipes/b2/standard/conanfile.py:50]:
```suggestion
        if self.options.toolset == 'cxx' or self.options.toolset == 'cross-cxx' and \
           not self.options.use_cxx_env:
            raise ConanInvalidConfiguration("Option toolset 'cxx' and 'cross-cxx' requires 'use_cxx_env=True'")
```

I prefer be more explicit.

File: recipes/b2/standard/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            'tru64cxx', 'vacpp', 'vc12', 'vc14', 'vc141', 'vc142' ]
+    }
+    default_options = {
+        'use_cxx_env': False,
+        'toolset': 'auto'
+    }
+
+    def configure(self):
+        if self.options.toolset == 'cxx' or self.options.toolset == 'cross-cxx' and \
+           not self.options.use_cxx_env:
```
Comment (madebr) [inline recipes/b2/standard/conanfile.py:50]: Shouldn't the *or* have parentheses?
```suggestion
        if (self.options.toolset == 'cxx' or self.options.toolset == 'cross-cxx') and \
                not self.options.use_cxx_env:
```

**Comment by Minimonium** [comment]
Thanks for the PR! It fixed the https://docs.conan.io/en/latest/howtos/sanitizers.html?highlight=sanitizer#using-from-custom-profiles user story for us.

**Pull request merged**